variable/test/operations_test.cpp

```diff
@@ -184,9 +184,9 @@ TEST(Variable, operator_plus_unit_fail) {
   auto b = makeVariable<double>(Dims{Dim::X}, Shape{2}, Values{1.0, 2.0},
                                 Variances{3.0, 4.0});
   b.setUnit(units::s);
-  ASSERT_ANY_THROW(a + b);
+  ASSERT_ANY_THROW([[maybe_unused]] auto c = a + b);
   b.setUnit(units::m);
-  ASSERT_NO_THROW(a + b);
+  ASSERT_NO_THROW([[maybe_unused]] auto c = a + b);
```

Comment:
Note that we have these helpers for this purpose: https://github.com/scipp/scipp/blob/92266385399dec2b40ad6bccf1a726db766619b6/test/test_macros.h#L55-L61